### PR TITLE
 Issue #183 - Address new vulnerabilities found in SCTP-AUTH

### PR DESCRIPTION
**Pull request by emanjon**
First commit of several to this PR

### CODE REVIEW

**Review by teiclap — APPROVED** [review]

**Comment by emanjon** [comment]
#184

File: draft-ietf-tsvwg-dtls-over-sctp-bis.md [hunk 1]
```diff
@@ -636,15 +644,26 @@ terminated and the associated keying material discarded.
...
+   the Transmission Sequence Numbers (TSNs) are unique. In fact, this
    document's solution is dependent on SCTP-AUTH and SCTP to prevent
    reordering, duplication, and removal of the DTLS records within
    each protected user message.  This includes detection of changes to
    what DTLS records start and end the SCTP user message, and removal of
    DTLS records before an increment to the epoch.  Without SCTP-AUTH,
    these would all have required explicit handling.
 
+   To prevent replay of DATA or I-DATA chunks resulting in impact on the received protected
+   user message, implementations MUST NOT cycle back to the Initial TSN and MUST
```
Comment (gloinul) [inline draft-ietf-tsvwg-dtls-over-sctp-bis.md:657]: So the SCTP stack will cycle back on the TSN. It is the SCTP-AUTH key that MUST be retired before 2^32 TSN number from the first used with this key.

File: draft-ietf-tsvwg-dtls-over-sctp-bis.md [hunk 2]
```diff
@@ -1012,47 +1036,58 @@ terminated and the associated keying material discarded.
...
-   is derived using the TLS-Exporter described in {{RFC5705}}.
+   Whenever a new DTLS connection is established, two 64-byte endpoint pair shared secrets
+   are derived using the TLS-Exporter described in {{RFC5705}}.
 
    After sending the DTLS Finished message
    for the initial DTLS connection, the active SCTP-AUTH key MUST be
-   switched from key identifier 0 to key identifier 1. Once the
+   switched from key identifier 0 to key identifiers 2 and 3. Once the
    initial Finished message from the peer has been processed by DTLS,
    the SCTP-AUTH Shared Key Identifier zero MUST NOT be used.
```
Comment (gloinul) [inline draft-ietf-tsvwg-dtls-over-sctp-bis.md:1072]: Should the "zero (0)" be used to make it clear when one talk about key identifier 0 above?

File: draft-ietf-tsvwg-dtls-over-sctp-bis.md [hunk 3]
```diff
@@ -1012,47 +1036,58 @@ terminated and the associated keying material discarded.
...
 
-   TBD.... Finished and close_notify works differently in DTLS 1.3
+   After sending or receiving the DTLS server Finished message
+   for the initial DTLS connection, the active SCTP-AUTH key MUST be
+   switched from key identifier 0 to key identifiers 2 and 3 and
+   the SCTP-AUTH Shared Key Identifier zero MUST NOT be used.
+
+   When the endpoint has sent or received a close_notify in one direction
+   on the old DTLS connection then the endpoint SHALL remove the SCTP-AUTH endpoint
+   pair shared secret associated with that direction in the old DTLS connection.
```
Comment (gloinul) [inline draft-ietf-tsvwg-dtls-over-sctp-bis.md:1090]: Should the requirement to retire the key before the TSN has wrapped once from the first TSN that contained a usage of this write SCTP-AUTH key be added here?

**Review by gloinul — COMMENTED** [review]
Some clarifications regarding key lifetime needed. Otherwise it looks good.

**Comment by gloinul** [comment]
This sentence in Parallel DTLS connection needs to be fixed: "here is also only a single SCTP-AUTH key exported per DTLS connection ensuring that there is clear mapping between the DTLS connection ID and the SCTP-AUTH security context for each key-id."

File: draft-ietf-tsvwg-dtls-over-sctp-bis.md (hunk 2 above)
Comment (emanjon) [reply draft-ietf-tsvwg-dtls-over-sctp-bis.md:1072]: fixed

**Comment by emanjon** [comment]
EXPORTER-DTLS-OVER-SCTP-CLIENT-WRITE
 EXPORTER-DTLS-OVER-SCTP-SERVER-WRITE

or 

 EXPORTER-DTLS-OVER-SCTP-CLIENT-SEND
 EXPORTER-DTLS-OVER-SCTP-SERVER-SEND
 
 (write and read keys is the terminology used by TLS)

**Review by teiclap — APPROVED** [review]

**Pull request merged**